kubernetes/resource_kubernetes_runtime_class_v1.go

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package kubernetes
+
+import (
+	"context"
+	"log"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	nodev1 "k8s.io/api/node/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgApi "k8s.io/apimachinery/pkg/types"
+)
+
+func resourceKubernetesRuntimeClassV1() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceKubernetesRuntimeClassV1Create,
+		ReadContext:   resourceKubernetesRuntimeClassV1Read,
+		UpdateContext: resourceKubernetesRuntimeClassV1Update,
+		DeleteContext: resourceKubernetesRuntimeClassV1Delete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"metadata": metadataSchema("runtimeclass", true),
+
+			"handler": {
+				Type:         schema.TypeString,
+				Description:  "Specifies the underlying runtime and configuration that the CRI implementation will use to handle pods of this class",
+				Required:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[a-z0-9]([-a-z0-9]*[a-z0-9])?`), ""),
```

Comment:
Kubernetes actually has a utility function to validate RFC 1123 DNS labels: https://pkg.go.dev/k8s.io/apimachinery/pkg/util/validation#IsDNS1123Label


So we don't need to roll our own regex here.